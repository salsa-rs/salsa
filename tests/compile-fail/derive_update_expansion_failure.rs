#[derive(salsa::Update)]
union U {
    field: i32,
}

#[derive(salsa::Update)]
struct S {
    #[update(with(missing_unsafe))]
    bad: i32,
}

fn missing_unsafe(_: *mut i32, _: i32) -> bool {
    true
}

fn main() {}
