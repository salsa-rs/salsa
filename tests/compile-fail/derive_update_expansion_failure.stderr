error: `derive(Update)` does not support `union`
 --> tests/compile-fail/derive_update_expansion_failure.rs:2:1
  |
2 | union U {
  | ^^^^^

error: expected `unsafe`
 --> tests/compile-fail/derive_update_expansion_failure.rs:8:14
  |
8 |     #[update(with(missing_unsafe))]
  |              ^^^^
